### PR TITLE
Responsive layout and navigation

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Implements most of the core responsive UI functionality; the diff was starting to get randomized so I decided to stop here for a PR to hopefully make reviewing a little easier. 

Screen Shot 2021-02-23 at 12 03 49 PM

This PR contains: 

- Basic responsive layout of header, footer, and narrative pages; nothing should be completely unusable on small screens anymore
- Header contains a collapsible navigation menu on small screens; this employs the same library we used in v1

mobile menu

Known issues (to be addressed in subsequent PRs) include: 
- mobile dropdown menus are not yet implemented
- the Sentence Types chart is the only one whose mobile styles did not survive the v2 transition; need to fix that
- the scroll-snapping behavior was already pretty janky and it feels even jankier to me on mobile

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Part of #309 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: spotlight-client/src/SiteNavigation/SiteNavigation.test.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import SiteNavigationContainer from ".";
+
+jest.mock("@w11r/use-breakpoint");
+
+const useBreakpointMock = useBreakpoint as jest.Mock;
+
+beforeEach(() => {
+  useBreakpointMock.mockReturnValue(true);
+  runInAction(() => {
+    DataStore.tenantStore.currentTenantId = "US_ND";
```
Comment (jovergaag) [inline spotlight-client/src/SiteNavigation/SiteNavigation.test.tsx:33]: I did notice that the behavior here is confusing when there isn't a tenant selected and the menu button is clicked.. I'm sure support for that is coming, just wanted to flag it.

**Review by jovergaag — COMMENTED** [review]
Looks good!

There are a couple of sections where the chart is so wide that it squeezes the narrative to an awkward degree. Will this be addressed in a following styling PR?

(demo data)
Screen Shot 2021-02-24 at 11 37 23 AM

File: spotlight-client/src/SiteNavigation/SiteNavigation.test.tsx (hunk 1 above)
Comment (macfarlandian) [reply spotlight-client/src/SiteNavigation/SiteNavigation.test.tsx:33]: yeah at the moment if you do that the menu won't have anything in it ... that will change eventually

**Comment by macfarlandian** [comment]
@jovergaag I did realize somewhat belatedly that tablet sizes are still awkward, as you point out ... I do have a fix in the works but it's going to come in the next PR, it's kind of wrapped up with the scrolling behavior and some other stuff also. sorry this is not very organized! I just kind of keep circling around and finding new problems, you know how it goes

**Review by jovergaag — APPROVED** [review]
🖌️

**Pull request merged**